pyanaconda/ui/gui/spokes/installation_source.py

```diff
@@ -1611,12 +1650,15 @@ def on_repoStore_row_inserted(self, model, path, itr, user_data=None):
                                                                              Gtk.TreeRowReference.new(model, path)))
 
     def on_repoProtocolComboBox_changed(self, combobox, user_data=None):
-        # Set the mirrorlist and proxy fields sensitivity depending on whether NFS was selected
-        sensitive = not(self._repoProtocolComboBox.get_active_id() == PROTOCOL_NFS)
-        fancy_set_sensitive(self._repoMirrorlistCheckbox, sensitive)
-        fancy_set_sensitive(self._repoProxyUrlEntry, sensitive)
-        fancy_set_sensitive(self._repoProxyUsernameEntry, sensitive)
-        fancy_set_sensitive(self._repoProxyPasswordEntry, sensitive)
+        # Set the url type and proxy fields sensitivity depending on whether NFS was selected
+        protocol = self._repoProtocolComboBox.get_active_id()
+        can_have_proxy = protocol in (PROTOCOL_HTTP, PROTOCOL_HTTPS, PROTOCOL_FTP,
+                                      PROTOCOL_MIRROR)
+        fancy_set_sensitive(self._repoProxyUrlEntry, can_have_proxy)
+        fancy_set_sensitive(self._repoProxyUsernameEntry, can_have_proxy)
+        fancy_set_sensitive(self._repoProxyPasswordEntry, can_have_proxy)
+        can_have_mirror = protocol in (PROTOCOL_HTTP, PROTOCOL_HTTPS)
+        fancy_set_sensitive(self._repoUrlTypeComboBox, can_have_mirror)
```

Comment:
It could be nice to insert blank lines before the `can_have_proxy` and `ca_have_mirror` to visually separate these parts.

Reply:
Agreed, I originally have them in.